src/pytorch_lightning/strategies/colossalai.py

```diff
@@ -0,0 +1,275 @@
+from typing import Any, Generator, Optional
+
+import torch
+from lightning_utilities.core.imports import RequirementCache
+
+import pytorch_lightning as pl
+from pytorch_lightning.accelerators.cuda import CUDAAccelerator
+from pytorch_lightning.overrides.base import _LightningModuleWrapperBase, _LightningPrecisionModuleWrapperBase
+from pytorch_lightning.plugins.precision import ColossalAIPrecisionPlugin
+from pytorch_lightning.strategies.ddp import DDPStrategy
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+
+_COLOSSALAI_AVAILABLE = RequirementCache("colossalai")
+if _COLOSSALAI_AVAILABLE:
+    from colossalai.context import ParallelMode
+    from colossalai.core import global_context as gpc
+    from colossalai.gemini import ChunkManager, GeminiManager
+    from colossalai.logging import disable_existing_loggers, get_dist_logger
+    from colossalai.nn.optimizer import CPUAdam, HybridAdam
+    from colossalai.nn.parallel import ZeroDDP
+    from colossalai.tensor import ProcessGroup
+    from colossalai.utils import get_current_device
+    from colossalai.utils.model.colo_init_context import ColoInitContext
+    from colossalai.zero import ZeroOptimizer
+
+
+class ModelShardedContext(ColoInitContext):
+    def _post_init_method(self, module: torch.nn.Module, *args, **kwargs):
+        if getattr(module, "_colossalai_module", False) is True:
+            return
+        super()._post_init_method(module, *args, **kwargs)
+        module._colossalai_module = True
+
+
+class ColossalAIStrategy(DDPStrategy):
```

Comment:
also need to add it to the registry so that users can pass `Trainer(strategy='collosalai')`.
Check deepspeed `register_strategies` for reference.

Reply:
Done.